### PR TITLE
Planet Switch

### LINKED ISSUE #63: add settings for turning planets on/off for spawn and for goal

### PR DESCRIPTION
**Pull request by bucketon**
- [x] refactor all spawn point list filtration logic somewhere
- [x] figure out a better solution than mapping enum fields to menu strings
- [x] fix bug where the nomai grand prix menu doesn't get refreshed when the settings are changed
- [x] add enum values for quantum moon and white hole

### CODE REVIEW

**Comment by bucketon** [comment]
https://github.com/bucketon/OWRandomSpeedrun/issues/63

File: NomaiGrandPrix/NomaiGrandPrix.cs [hunk 1]
```diff
@@ -191,8 +208,13 @@ private void SpeedRunButton_OnClick()
 
         private void ResetRunButton_OnClick()
         {
-            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { return config.shouldSpawn; });
-            SpeedrunState.GoalPoint = GetRandomSpawnConfig(config => { return config.shouldGoal; });
+            //TODO: figure out a good place to consolidate this logic since it's duplicated 4 times.
+            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { 
+              return config.shouldSpawn && (config.area == Area.None || ModHelper.Config.GetSettingsValue<bool>($"Spawn{areaNameMap[config.area]}"));
```
Comment (cmaines) [inline NomaiGrandPrix/NomaiGrandPrix.cs]: We could use [Enum.Parse()](https://learn.microsoft.com/en-us/dotnet/api/system.enum.parse?view=net-7.0) here. Something like `$"Spawn{Enum.Parse(typeof(Area), config.area)}"`. That would let us get rid of the static map.

File: NomaiGrandPrix/NomaiGrandPrix.cs [hunk 2]
```diff
@@ -191,8 +208,13 @@ private void SpeedRunButton_OnClick()
 
         private void ResetRunButton_OnClick()
         {
-            SpeedrunState.SpawnPoint = GetRandomSpawnConfig(config => { return config.shouldSpawn; });
-            SpeedrunState.GoalPoint = GetRandomSpawnConfig(config => { return config.shouldGoal; });
+            //TODO: figure out a good place to consolidate this logic since it's duplicated 4 times.
```
Comment (cmaines) [inline NomaiGrandPrix/NomaiGrandPrix.cs:211]: It looks like it's only being used in this class; would a private function encapsulating it be sufficient?

File: NomaiGrandPrix/NomaiGrandPrix.cs (hunk 1 above)
Comment (bucketon) [reply NomaiGrandPrix/NomaiGrandPrix.cs]: good call. I'll add that and change the naming so that it works.

Comment (bucketon) [reply NomaiGrandPrix/NomaiGrandPrix.cs]: fixed

File: NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs [hunk 3]
```diff
@@ -355,7 +361,7 @@ private void SetInitialSelection(SpawnPointMenu menu, SpawnPointList list)
 
         private Selectable GetRandomSelectable(SpawnPointMenu menu)
         {
-            var randomIndex = _random.Next(_fromMenu._menuOptions.Length);
+            var randomIndex = _random.Next(menu._menuOptions.Length);
```
Comment (cmaines) [inline NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs:364]: good catch, thx

File: NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs [hunk 4]
```diff
@@ -219,6 +220,11 @@ public void OnItemClicked(PointerEventData eventData, SpawnPointMenuOption optio
             }
         }
 
+        public void Refresh()
+        {
+            Destroy(gameObject);
```
Comment (cmaines) [inline NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs:225]: What's causing this to be re-initialized?

Comment (bucketon) [reply NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs:225]: `_manager = SpawnPointSelectorManager.Instance;` in NomaiGrandPrix.cs ln 237

Comment (cmaines) [reply NomaiGrandPrix/SpawnPointMenu/SpawnPointSelectorManager.cs:225]: Ah got it, thought this was NomaiGrandPrix before.

**Review by cmaines — APPROVED** [review]

**Pull request merged**